### PR TITLE
feat: appflow ingestion stage

### PR DESCRIPTION
**Pull request by malachi-constant**
### Feature or Bugfix
- Add stage `AppFlowIngestionStage`
- Adds 100% testing
- CDK `2.60.0` -> `2.61.0`

### Relates
- #215 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/stages/appflow-ingestion.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    const { flowName, flowExecutionStatusCheckPeriod, destinationFlowConfig, sourceFlowConfig, flowTasks } = props;
+
+    if (!flowName) {
+      // Check required props for CfnFlow create and except if not provided
+      if (!destinationFlowConfig || !sourceFlowConfig || !flowTasks) {
+        throw new Error(
+          "if 'flowName' is not specified, 'destinationFlowConfig', 'sourceFlowConfig' & 'tasks' are required properties",
+        );
+      }
+      const flow = new appflow.CfnFlow(this, `${id}-flow`, {
```
Comment (LeonLuttenberger) [inline src/stages/appflow-ingestion.ts:38]: You should just make `"Flow"` the ID for this resource. The ID will implicitely be prefixed the parent construct's ID anyway, so this way we're just duplicating it.

File: src/stages/appflow-ingestion.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        destinationFlowConfigList: [destinationFlowConfig],
+        flowName: `${id}-flow`,
```
Comment (LeonLuttenberger) [inline src/stages/appflow-ingestion.ts:40]: I think we should remove the explicit `flowName`. That way CFN will just generate the name.

File: src/stages/appflow-ingestion.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["appflow:StartFlow"],
+        resources: ["*"],
+      }),
+    );
+  }
+
+  private createStartFlowCustomTask(flowName: string): sfn.CustomState {
+    const stateJson = {
```
Comment (LeonLuttenberger) [inline src/stages/appflow-ingestion.ts:91]: We could probably make this cleaner by using the [CallAwsService](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-stepfunctions-tasks.CallAwsService.html) task. But we can save that for a separate PR.

File: src/stages/appflow-ingestion.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+          Next: "wait-before-checking-flow-status-again",
+        },
+      ],
+    };
+
+    return new sfn.CustomState(this, "start-flow-execution", { stateJson });
+  }
+
+  private createCheckFlowExecutionTask(id: string): tasks.LambdaInvoke {
+    const statusLambdaRole = new iam.Role(this, `{id}-flow-execution-status-lambda-role`, {
```
Comment (LeonLuttenberger) [inline src/stages/appflow-ingestion.ts:109]: The ID can just be `"Flow Execution Status Lambda Role"`.

File: src/stages/appflow-ingestion.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      assumedBy: new iam.ServicePrincipal("lambda.amazonaws.com"),
+      description: "lambda role to check appflow flow execution status",
+    });
+
+    const statusLambda = new lambda.Function(this, `${id}-flow-execution-status-lambda`, {
```
Comment (LeonLuttenberger) [inline src/stages/appflow-ingestion.ts:114]: The ID can just be `"Flow Execution Status Lambda"` :)

File: src/stages/appflow-ingestion.ts (hunk 1 above)
Comment (malachi-constant) [reply src/stages/appflow-ingestion.ts:38]: Right, and this is something we are doing across most of the DDK in its current state I believe.

**Review by LeonLuttenberger — APPROVED** [review]

File: src/stages/appflow-ingestion.ts (hunk 3 above)
Comment (malachi-constant) [reply src/stages/appflow-ingestion.ts:91]: Will handle in #223

**Pull request merged**